hudi-flink/src/main/java/org/apache/hudi/util/StringToRowDataConverter.java

```diff
@@ -85,7 +87,8 @@ private static Converter getConverter(LogicalType logicalType) {
         // see HoodieAvroUtils#convertValueForAvroLogicalTypes
         return field -> (int) LocalDate.parse(field).toEpochDay();
       case TIMESTAMP_WITHOUT_TIME_ZONE:
-        return field -> TimestampData.fromEpochMillis(Long.parseLong(field));
+        // see HoodieAvroUtils#convertValueForAvroLogicalTypes
+        return field -> TimestampData.fromEpochMillis(LocalDateTime.parse(field).atZone(ZoneId.systemDefault()).toInstant().toEpochMilli());
```

Comment:
Is this change backwards compatible ?

Reply:
Why we need this change, is it related with this patch ? And the change is wrong because the string format is already in epoch milliseconds but not the format `DateTimeFormatter.ISO_LOCAL_DATE_TIME`.

Reply:
Yes it is related to this patch. In HoodieAvroUtils, whenever the LogicalType is timestamp, LocalDateTime instance is being returned instead of Long. Hence in string form, we would get a DateTime string in StringToRowConverter instead of stringified Long as before. Hence it needs to be parsed using LocalDateTime parser. Check the handling of LocalDate in the line above which was there from before because LogicalType date was handled correctly in HoodieAvroUtils before only.